src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -347,67 +331,33 @@ object NamespaceWithWorkflow {
 
     val workflow: Workflow = Scope.generateWorkflow(workflowAst, namespaces, tasks, wdlSyntaxErrorFormatter)
 
-    // FIXME: This block is run for its side effect of blowing up on the .get (I believe!) - Should there be a real syntax error?
-    // FIXME: It took me a while to understand the logic of the original code & I'm not sure this comment is correct?
-    // FIXME: Also, it'd be nice to move this validity check into Workflow if possible
-    // All MemberAccess ASTs that are not contained in other MemberAccess ASTs
+    // Make sure that all MemberAccess ASTs refer to valid calls and those have valid output names
    for {
       call <- workflow.calls
       (name, expression) <- call.inputMappings
-      memberAccess <- expression.ast.findTopLevelMemberAccesses()
-    } yield {
-      getCallFromMemberAccessAst(memberAccess, workflow, wdlSyntaxErrorFormatter).get
-    }
+      memberAccessAst <- expression.ast.findTopLevelMemberAccesses()
+    } validateMemberAccessAst(memberAccessAst, workflow, wdlSyntaxErrorFormatter)
```

Comment:
Yeah these changes didn't make the situation worse or better, it just made me look at a piece of code that I think was meant to be quick-n'-dirty at the time it was written but seems to have lingered.

I also did just notice this `for` has no `yield`.  Interesting...